tmt/checks/watchdog.py

```diff
@@ -0,0 +1,467 @@
+import dataclasses
+import datetime
+import re
+import threading
+import time
+from collections.abc import Iterable
+from typing import TYPE_CHECKING, Optional
+
+import tmt.log
+import tmt.steps.execute
+import tmt.steps.provision
+import tmt.steps.provision.artemis
+import tmt.steps.provision.connect
+import tmt.steps.provision.local
+import tmt.steps.provision.mrack
+import tmt.steps.provision.podman
+import tmt.steps.provision.testcloud
+import tmt.utils
+from tmt.checks import Check, CheckPlugin, provides_check
+from tmt.result import CheckResult, ResultOutcome
+from tmt.utils import Path, field, render_run_exception_streams
+
+if TYPE_CHECKING:
+    from tmt.steps.execute import TestInvocation
+
+PING_OUTPUT_PATTERN = re.compile(
+    r'(?m)(?P<transmitted>\d+) packets transmitted, (?P<received>\d+) received')
+SSH_PING_OUTPUT_PATTERN = re.compile(r'Ncat: Connected')
+
+PINGABLE_GUEST_CLASSES: tuple[type[tmt.steps.provision.Guest], ...] = (
+    tmt.steps.provision.artemis.GuestArtemis,
+    tmt.steps.provision.connect.GuestConnect,
+    tmt.steps.provision.mrack.GuestBeaker,
+    # TODO: is there a way to ping the VM instead of localhost?
```

Comment:
@happz is there an issue opened for this? would be a major drawback not to be able use watchdog with these hosts,

Reply:
See https://github.com/teemtee/tmt/pull/2412#discussion_r1513013104 - I'll add a comment since it's already two people who asked about it, but, I have no issue tracking this, and it's possible to use "SSH ping" with this guest. I plan to extend support to other guests as well, but I suppose have to get used to that some features will not work everywhere, e.g. ping will not work with the `container` plugin, I suppose, not without constructing custom network topologies, and we're nowhere near that. And I gues that would be the case of `virtual` as well.

Reply:
Added in c3d9d9f9b9df3494f456ee5ff025200b478f0b60

Reply:
thank you!